metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -275,77 +295,101 @@ final class BloopServers(
    */
   def ensureDesiredJvmSettings(
       maybeRequestedBloopJvmProperties: Option[List[String]],
```

Comment:
I think the issue with test is coming from the fact, that we are sending `None` here and we already have something set for CI. None should be treated as `use whatever is set currently` while `Some(Nil)` would be `remove all the options`

Reply:
Thank you for this. I think the guard 
```
bloopJsonUpdateCause <-
          if (
            maybeRequestedBloopJvmProperties != maybeBloopGlobalJsonJvmProperties
          ) Some(BloopJsonUpdateCause.JVM_OPTS)
          else if (maybeRequestedMetalsJavaHome != maybeBloopGlobalJsonJavaHome)
            Some(BloopJsonUpdateCause.JAVA_HOME)
          else None

```
was enforcing that! Still, as an extra test, I checked that the `maybeRequestedBloopJvmProperties` is never `None`
It can only be `List.empty` when the user has not set anything!

Reply:
I think it would be enought to add and if:
`if maybeRequestedBloopJvmProperties.nonEmpty`
somewhere in the for comprehension

Reply:
Then what about the cases when it’s empty, while javaHome is set?

Reply:
I think it might have to do with how the language client is called.

Reply:
specifically in cases when the bloop global json is manually premodified. The errors do not show for the other case.